android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java

```diff
@@ -46,12 +63,118 @@ public void setUp() {
+                .perform(replaceText(date), closeSoftKeyboard());
+
+        onView(allOf(withId(com.google.android.material.R.id.confirm_button), isDisplayed()))
+                .perform(click());
+
+        onView(withId(com.google.android.material.R.id.material_timepicker_cancel_button))
+                .perform(click());
+
+        onView(withId(R.id.date_text)).check(matches(withText(date)));
+
+
+    }
+
+
+    @Test
+    public void testDatePickerTimePicker() {
+
+        String date = "24.05.2023, 16:50 Uhr";
+
+        onView(withId(R.id.date_layout))
+                .perform(AddingFragmentTest.clickIcon(true));
+
+        onView(withId(com.google.android.material.R.id.mtrl_picker_header_toggle))
+                .perform(click());
+
+        onView(allOf(TestMatchers.childAtPosition(TestMatchers.childAtPosition(withId(com.google.android.material.R.id.mtrl_picker_text_input_date),
+                0), 0), isDisplayed()))
+                .perform(replaceText(date), ViewActions.closeSoftKeyboard());
+
+        onView(allOf(withId(com.google.android.material.R.id.confirm_button), isDisplayed()))
+                .perform(click());
+
+        onView(allOf(TestMatchers.childAtPosition(TestMatchers.childAtPosition(withClassName(is("com.google.android.material.textfield.TextInputLayout")),
+                0), 0), isDisplayed()))
+                .perform(replaceText("16"));
+
+        onView(withId(com.google.android.material.R.id.material_minute_text_input)).perform(click());
+
+        onView(allOf(TestMatchers.childAtPosition(TestMatchers.childAtPosition(withClassName(is("com.google.android.material.textfield.TextInputLayout")),
+                0), 0), isDisplayed()))
+                .perform(replaceText("50"), closeSoftKeyboard());
+
+        onView(allOf(withId(com.google.android.material.R.id.material_timepicker_ok_button)))
+                .perform(click());
+
+        onView(withId(R.id.date_text))
+                .check(matches(withText(date)));
+
+
+    }
+    public static ViewAction clickIcon(boolean isEndIcon) {
+        return new ViewAction() {
+
+            @Override
+            public Matcher<View> getConstraints() {
+                return ViewMatchers.isAssignableFrom(TextInputLayout.class);
+            }
+
+            @Override
+            public String getDescription() {
+                return "Clicks the end or start icon";
+            }
+
+            @Override
+            public void perform(UiController uiController, View view) {
+                TextInputLayout item = (TextInputLayout) view;
+                // Reach in and find the icon view since we don't have a public API to get a reference to it
+                CheckableImageButton iconView =
+                        item.findViewById(isEndIcon ? com.google.android.material.R.id.text_input_end_icon : com.google.android.material.R.id.text_input_start_icon);
+                iconView.performClick();
+            }
+        };
```

Comment:
This could be part of a `TestViewActions` class. In that case you should rename the method to `ClickTextInputLayoutIcon` as it only works for TextInputLayouts